wire/ARPHeader.h

```diff
@@ -0,0 +1,55 @@
+/* vim: set expandtab ts=4 sw=4: */
+/*
+ * You may redistribute this program and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef ARPHeader_H
+#define ARPHeader_H
+
+#include "util/Assert.h"
+#include "util/Endian.h"
+#include "wire/Ethernet.h"
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+struct ARPHeader_pre {
+    uint16_t htype;
+    uint16_t ptype;
+    uint8_t hlen;
+    uint8_t plen;
+    uint16_t operation;
+};
```

Comment:
```c
#define ARPHeader_Pre_SIZE 8
Assert_compileTime(sizeof(struct ARPHeader_Pre) == ARPHeader_Pre_SIZE);
```
In general this is to catch weird compilers which might decide to pad inside of the structure to align the fields.